### PR TITLE
Upgrade pdc sdk

### LINKED ISSUE #836: Upgrade `@pdc/sdk`
We need to update the front-end to use the latest SDK produced by the back-end. Since it contains breaking changes, we'll need to update our client code to match.

See also PR #832

### PR DESCRIPTION
**Pull request by hminsky2002**
This PR upgrades the sdk to the latest version, and refactors the codebase to reflect two major updates: the change in name from the entity 'organization' to 'changemaker', and the requirement for a source id to be referenced when creating a new bulkupload. Notably, the current approach to providing a source is limited to a constant number, wherein we fetch the first 1000 sources in the database and filter only on that. See [service issue 1295](https://github.com/PhilanthropyDataCommons/service/issues/1295). 

Closes #836

### CODE REVIEW

**Comment by bickelj** [comment]
@hminsky2002 I added the above deploy preview address to the valid redirect URIs for the client associated with the front end, so now login should work there. And when looking at the preview, I see the Changemakers and the Proposals. Nice.

File: src/components/AppNavbar.tsx
Comment (bickelj) [inline src/components/AppNavbar.tsx:33]: Typo here and confirmed it breaks the link by looking at deploy preview.

File: src/pages/AddData.tsx
Comment (bickelj) [inline src/pages/AddData.tsx:64]: Good call throwing an `Error` here. I wonder how it surfaces and if the message could be more explicit, like `No System Source with short code = yada found at [url used]`.

File: src/utils/sources.ts
Comment (bickelj) [inline src/utils/sources.ts:12]: Unfortunate that this needs to be done but a clean and concise implementation given that it needs to be done!

**Review by bickelj — APPROVED** [review]
I was able to see the app in action at the deploy preview. I did not try to run it locally, however. The changes all appear expected to me.

Since the app is broken until this gets merged, I am smashing "Approve" even though I know near-zero React. If you'd like to wait for others feedback, I cannot disagree with that decision either.

My simple way of thinking is that by merging this PR, if something breaks it won't be significantly more broken than it is now, and we have an opportunity to unbreak it with a merge.

File: package.json
Comment (jasonaowen) [inline package.json:9]: Right now, this upgrade happens in the first commit, and therefore the first commit does not build.

Instead, let's gradually bump the SDK version along with the changes necessary for that specific version of the SDK. I believe, in order, they are: handle timestamp strings, include a source id with bulk uploads, and rename organizations to changemakers.

File: src/pdc-api.ts
Comment (jasonaowen) [inline src/pdc-api.ts:221]: This works, but the API docs do not list the `_page` and `_count` parameters. New issue opened: https://github.com/PhilanthropyDataCommons/service/issues/1313

File: src/pages/AddData.tsx
Comment (jasonaowen) [inline src/pages/AddData.tsx:41]: The problem that @hminsky2002 called out with this is that if the system source does not show up in this call, we error down below.

This is exasperated by two factors: using a synthetic key so that we need to do this at all, and that the `/sources` endpoint returns the most recent sources (sorted by id descending) rather than the earliest sources, as the system source should always be the first source.

The change that introduced the system source, https://github.com/PhilanthropyDataCommons/service/pull/1116, created the table and inserted the system source row in the same migration, and the id is `GENERATED ALWAYS AS IDENTITY`, so the system source ID should always be `1` and cannot be `UPDATE`d. Maybe it's worth hardcoding that here? cc @slifty 

```psql
pdc=*> update sources set id = 2;
ERROR:  column "id" can only be updated to DEFAULT
DETAIL:  Column "id" is an identity column defined as GENERATED ALWAYS.
```

**Review by jasonaowen — CHANGES_REQUESTED** [review]
Looking good! Thanks for working on this, @hminsky2002.

I believe this covers what we discussed during my office hour.

File: package.json
Comment (hminsky2002) [reply package.json:9]: @jasonaowen Egads! I've hit an unfortunate issue here -- the SourceBundle, which is needed for the source handling, was missed in the openapi spec and I had to add it back in (https://github.com/PhilanthropyDataCommons/service/pull/1281) at a later date. As such, an older version of the pdc cannot be used for the sourceId commit without also refactoring many organizations to changemakers. Thoughts on either merging the two commits? That seems messy but then again, the changes are separate enough that they might still be discnerable

Comment (slifty) [reply package.json:9]: My .02 which may not be agreed on here is that it is appropriate to have one commit that bumps to the latest version along with all required changes to make that bump.

I know this is not ideal, but also I think this is the price of allowing versions to bump so much!

Alternatively Harry you could decide to take Jason's approach and bump one version at a time.  I'm just not sure that's worth the extra dev time really.

File: src/pdc-api.ts
Comment (slifty) [reply src/pdc-api.ts:221]: I'll fix this up thanks for the catch @jasonaowen

File: src/pages/AddData.tsx
Comment (slifty) [reply src/pages/AddData.tsx:41]: @jasonaowen that's fine; in the long run IMHO it would be appropriate to provide a documented API mechanism to load the system source rather than relying on magic values.

Comment (jasonaowen) [reply src/pages/AddData.tsx:41]: @slifty that could be good, although it's not clear to me how often it'd be used or if it'd be used by other partners or only for the type of admin work that our team is still forecast to do. If it's just us, a magic number is fine, and `1` is the second least-magical number! If we could encode that in the SDK somehow, or at least document it, that would be good.

For now, @hminsky2002, I suggest a new `useSystemSource` hook that fetches `/sources/1`, so that you don't have to worry about search or pagination or anything like that.

**Comment by hminsky2002** [comment]
@jasonaowen @slifty In a delightful turn of events, since we are now only retrieving a single Source object for the systemSource, I was able to at least split this PR into source handling and changemaker refactor commits! Also, I added a small change to accommodate the recent refactor of the bulk-uploads route! Should be ready for final review/merging!

File: src/pages/AddData.tsx
Comment (jasonaowen) [inline src/pages/AddData.tsx:38]: I believe this should be using `useCallback`, because right now it appears to be recreating this callback on every render of the `<BulkUploader>` component.

However, this is not a new problem, so it doesn't need to be fixed in this PR.

**Review by jasonaowen — APPROVED** [review]
This looks great! Thanks for working through all that.

I was able to test it by successfully uploading a bulk CSV, and poking around in the changemakers tab.

**Comment by jasonaowen** [comment]
Just to add some breadcrumbs for later, this also handles the front-end portion of https://github.com/PhilanthropyDataCommons/service/issues/1117 .

**Pull request merged**
